src/abstract_charm.py

```diff
@@ -265,13 +280,7 @@ def reconcile(self, event=None) -> None:  # noqa: C901
                         shell=workload_.shell,
                     )
             if workload_.container_ready:
-                cos_relation_exists = (
-                    self._cos.relation_exists and not self._cos.is_relation_breaking(event)
-                )
-                workload_.reconcile(
-                    unit_name=self.unit.name,
-                    exporter_config=self._cos.exporter_user_info if cos_relation_exists else None,
-                )
+                workload_.reconcile_services(event)
```

Comment:
passing inputs is acceptable when calling `_reconcile` from abstract_charm.py. these inputs are values collected from tls and cos at the moment

however, i am not a fan of tls files import cos just so it can pass in the necessary inputs

solution proposed in this PR: 
- abstract_charm exposes properties that expose these inputs (since abstract_charm already can import all of the relevant files)
- pass a reference to the charm into the workload class, and use above properties to retrieve inputs

as an alternative, i am ok with tls files calling the properties on the charm class (as it will have the reference to the charm object) rather than a wrapper (`_reconcile_services`)